### PR TITLE
chore: use golangci lint

### LINKED ISSUE #981: add Github actions to lint go code
### What is the version of your ORAS CLI

1.0.0

### What would you like to be added?

Use [existing Github](https://github.com/golangci/golangci-lint) to do static code formatting scan for pull requests.

### Why is this needed for ORAS?

As the number of contributor grows, adding such a CI job will help unifying coding style.

### Are you willing to submit PRs to contribute to this feature?

- [ ] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by kranurag7**
**What this PR does / why we need it**:
The following are the places that's giving some errors as of now according to the linter. 
<details><summary> output of running golangci locally </summary>
<p>

```bash
$ golangci-lint run
cmd/oras/root/attach.go:94:22: Error return value of `cmd.MarkFlagRequired` is not checked (errcheck)
        cmd.MarkFlagRequired("artifact-type")
                            ^
cmd/oras/root/discover.go:191:13: Error return value is not checked (errcheck)
                        printJSON(ref)
                                 ^
cmd/oras/internal/display/print.go:119:9: Error return value is not checked (errcheck)
                        Print("Tagging", ref)
                             ^
cmd/oras/internal/display/print.go:139:9: Error return value is not checked (errcheck)
                        Print("Tagging", ref)
                             ^
cmd/oras/internal/option/packer_test.go:73:14: Error return value of `os.WriteFile` is not checked (errcheck)
        os.WriteFile(testFile, []byte(testContent), fs.ModePerm)
                    ^
cmd/oras/internal/option/remote.go:276:30: Error return value of `repo.SetReferrersCapability` is not checked (errcheck)
                repo.SetReferrersCapability(*opts.distributionSpec.referrersAPI)
                                           ^
cmd/oras/internal/option/remote_test.go:54:29: Error return value of `(*encoding/json.Encoder).Encode` is not checked (errcheck)
                        json.NewEncoder(w).Encode(testTagList)
                                                 ^
cmd/oras/internal/option/remote_test.go:236:28: Error return value of `(*encoding/json.Encoder).Encode` is not checked (errcheck)
                json.NewEncoder(w).Encode(testTagList)
                                         ^
internal/cache/target_test.go:180:12: Error return value of `w.Write` is not checked (errcheck)
                                w.Write(blob)
                                       ^
internal/graph/graph_test.go:142:14: Error return value of `memory.Push` is not checked (errcheck)
                memory.Push(ctx, descs[i], bytes.NewReader(blobs[i]))
                           ^
internal/graph/graph_test.go:258:14: Error return value of `memory.Push` is not checked (errcheck)
                memory.Push(ctx, descs[i], bytes.NewReader(blobs[i]))
                           ^
internal/graph/graph_test.go:356:14: Error return value of `memory.Push` is not checked (errcheck)
                memory.Push(ctx, descs[i], bytes.NewReader(blobs[i]))
```

</p>
</details> 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #981 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [x]  Do all new files have an appropriate license header?

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: false
```
Comment (qweeah) [inline .github/workflows/golangci-lint.yml:33]:
```suggestion
     - name: Checkout
        uses: actions/checkout@v3
      - name: Setup Go ${{ matrix.go-version }}
        uses: actions/setup-go@v4
        with:
          go-version: ${{ matrix.go-version }}
          check-latest: true

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    paths-ignore:
+      - 'docs/**'
+
+permissions:
+  contents: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
```
Comment (qweeah) [inline .github/workflows/golangci-lint.yml:27]:
```suggestion
    runs-on: ubuntu-latest
    strategy:
      matrix:
        go-version: ['1.20']
      fail-fast: true
```

**Review by qweeah — CHANGES_REQUESTED** [review]
Suggested to align with other actions.

Also this PR requires fixing of all detected issues otherwise the PR check will fail. @kranurag7 Do you want to do the fixing? If not have time I can add another PR to fix all the lint issues found.

**Comment by qweeah** [comment]
@kranurag7 Can you add signoff as is suggested in https://github.com/oras-project/oras/pull/986/checks?check_run_id=14956872091:

> - In your local branch, run: git rebase HEAD~3 --signoff
> - Force push your changes to overwrite the branch: git push --force-with-lease origin golangci

**Comment by TerryHowe** [comment]
I just rebased, so you are going to have to `git pull` and take care of DCO

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**